### PR TITLE
Add UMA forcefield model

### PR DESCRIPTION
**Pull request by ekurtulus**
Add's JAX port of the UMA model.

## Checklist

- [x] Tests added/updated and passing
- [x] Appropriate labels added to PR

### CODE REVIEW

File: jax_md/ff/uma/common/__init__.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2024 Google LLC
```
Comment (abhijeetgangan) [inline jax_md/ff/uma/common/__init__.py]: https://github.com/jax-md/jax-md/tree/main/jax_md/_nn move things here from ff folder.

**Comment by abhijeetgangan** [comment]
@codex review

**Comment by abhijeetgangan** [comment]
@codex review

**Comment by abhijeetgangan** [comment]
@codex review

**Comment by abhijeetgangan** [comment]
@codex review

**Comment by abhijeetgangan** [comment]
For bookkeeping - JAX was ran with:

```
'--xla_gpu_deterministic_ops=true'
jax.config.update('jax_default_matmul_precision', 'highest')
```
while all PyTorch models were ran with:
```
torch.use_deterministic_algorithms(True, warn_only=True)
torch.backends.cudnn.deterministic = True
torch.backends.cuda.matmul.allow_tf32 = False
torch.backends.cudnn.allow_tf32 = False
torch.set_float32_matmul_precision('highest')
```
Note: not all operations for torch were deterministic. Also, see for other comparison for JAX vs PyTorch https://github.com/ACEsuit/mace-jax/pull/15 in terms of error distribution.

Comparison vs UMA-S-1p1
uma_parity_uma_s_1p1
Comparison vs UMA-S-1p2
uma_parity_uma_s_1p2
